src/app/modules/feedback/feedback.service.ts

```diff
@@ -151,3 +157,103 @@ export const hasNoPreviousFeedback = (
     }
     return okAsync(true as const)
   })
+
+/**
+ * Inserts given admin feedback to the database.
+ * @param userId the userId of the admin that provided the feedback
+ * @param rating the feedback rating to insert (0 for thumbs down, 1 for thumbs up)
+ * @param comment the feedback comment to insert if available
+ * @returns ok(IAdminFeedbackSchema) if successfully inserted
+ * @returns err(DatabaseError) on database error
+ */
+export const insertAdminFeedback = ({
+  userId,
+  rating,
+  comment,
+}: {
+  userId: string
+  rating: number
+  comment?: string
+}) => {
+  return ResultAsync.fromPromise(
+    AdminFeedbackModel.create({
+      userId,
+      rating,
+      comment,
+    }),
+    (error) => {
+      logger.error({
+        message: 'Database error when creating admin feedback document',
+        meta: {
+          action: 'insertAdminFeedback',
+          userId,
+        },
+        error,
+      })
+
+      return new DatabaseError('Admin feedback could not be created')
+    },
+  )
+}
+
+/**
+ * Updates admin feedback in the database.
+ * Will use previous value if comment or rating are not passed into method
+ * @param feedbackId the id of the admin feedback to update
+ * @param userId the id of the admin
+ * @param comment the feedback comment to insert
+ * @param rating the feedback rating to insert (0 for thumbs down, 1 for thumbs up)
+ * @returns ok(IAdminFeedbackSchema) if successfully inserted
+ * @returns err(MissingAdminFeedbackError) if feedback document with the same feedbackId and userId is not found
+ * @returns err(DatabaseError) on database error
+ */
+export const updateAdminFeedback = ({
+  feedbackId,
+  userId,
+  comment,
+  rating,
+}: {
+  feedbackId: string
+  userId: string
+  comment?: string
+  rating?: number
+}) => {
+  return ResultAsync.fromPromise(
+    AdminFeedbackModel.findById(feedbackId),
+    (error) => {
+      logger.error({
+        message: 'Database error when querying for  admin feedback document',
+        meta: {
+          action: 'updateAdminFeedback',
+          feedbackId,
+        },
+        error,
+      })
+
+      return new DatabaseError('Admin feedback could not be found')
+    },
+  ).andThen((adminFeedback) => {
+    // Ensure that feedbackId and userId are the same as params
```

Comment:
/nit this comment might be better worded to explain why we need to ensure `userId === adminFeedback.userId`.

```suggestion
    // Only allow updating of own feedback
```

Reply:
Refactored here! [eb93502](https://github.com/opengovsg/FormSG/pull/6454/commits/eb9350216c394b8ebc259da973a6ffc10ca288bc) though slightly different wording